improver_tests/set_up_test_cubes.py

```diff
@@ -60,48 +62,131 @@ def construct_xy_coords(ypoints, xpoints, spatial_grid):
             Number of grid points required along the x-axis
         spatial_grid (str):
             Specifier to produce either a "latlon" or "equalarea" grid
+        grid_spacing (int or float):
+            Grid resolution (degrees or metres).
+        domain_corner (list of int or float):
```

Comment:
Suggest using mypy syntax for type definitions - see [code style guide](https://github.com/metoppv/improver/wiki/Code-Style-Guide#variable-types). This doesn't match a lot of the code, but there was a team decision to move over to this style. Apply similar changes throughout the PR.

Reply:
Modified in commit e2b5b6e6d2bfa21224fab93d7929aeb78614a18b.

I was unsure how to specify when a variable can be for example a list of ints or a list of floats (rather than a list which can contain a mixture of ints and floats which I believe is covered by Union) so I may have done that incorrectly.

Also, in the [code style guide for docstrings](https://github.com/metoppv/improver/wiki/Code-Style-Guide#docstrings) it says that "private methods do not always require complete doc-strings, particularly if their behaviour is very obvious and the arguments to the method have been defined in other doc-strings", so would you recommend I remove the args in the docstrings for `_create_y_x_arrays`, `_set_domain_corner` and `_default_grid` as they are defined in the docstring for `construct_xy_coords`?

Reply:
If floats are an acceptable input, it's ok to specify the type as float rather than needing to explicitly include int. This is in the [mypy docs](https://mypy.readthedocs.io/en/stable/duck_type_compatibility.html) under duck typing.
There are a few instances in e2b5b6e that take either a list or a tuple. It'd be neater to use the generic Sequence type for these.

I'd be happy with these short private functions (`_create_y_x_arrays` and the others you listed) having one-line summary docstrings rather than all the args/return detail.

Reply:
Modified in commit 7f0a5eca6ad1e1a673b68f6b3da01e5e2991aa82.

Left the return details for clarity but removed args.